.github/workflows/dispatch-yocto-build-deploy.yml

```diff
@@ -0,0 +1,124 @@
+name: "Test Yocto"
+
+on:
+  pull_request:
+    types: [opened, synchronize]
+    branches:
+      - "main"
+      - "master"
+  pull_request_target:
+    types: [opened, synchronize]
+    branches:
+      - "main"
+      - "master"
+
+  workflow_dispatch:
+    # you may only define up to 10 `inputs` for a `workflow_dispatch` event
+    inputs:
+      device-repo:
+        description: balenaOS device repository (owner/repo)
+        required: true
+        type: string
+        default: balena-os/balena-generic
+      device-repo-ref:
+        description: balenaOS device repository tag, branch, or commit to build
+        required: false
+        type: string
+        default: master
+      meta-balena-ref:
+        description: meta-balena ref if not the currently pinned version
+        required: false
+        type: string
+      yocto-scripts-ref:
+        description: balena-yocto-scripts ref if not the currently pinned version
+        required: false
+        type: string
+      machine:
+        description: yocto board name
+        required: true
+        type: string
+        default: generic-amd64
+      environment:
+        description: Select deploy environment
+        required: false
+        type: choice
+        options:
+          - >
+            {
+              "environment": "balena-staging.com",
+              "s3-bucket": "resin-staging-img",
+              "s3-region": "us-east-1",
+              "aws-subnet": "subnet-0d73c1f0da85add17",
```

Comment:
Are subnet and security group needed to deploy to S3?

Reply:
They are required for the current publish AMI functions.